docs/typescript/testing.md

```diff
@@ -351,3 +349,12 @@ await worker.runUntil(
   testEnv.workflowClient.execute(functionToTest, workflowOptions) // Throws WorkflowFailedError
 );
 ```
+
+### Testing with Jest
+
+Although these testing docs use [Mocha](https://mochajs.org/), you can also use [Jest](https://jestjs.io/).
+There are a couple of caveats for testing with Jest:
+
+1. The Temporal TypeScript SDK only supports Jest `>= 27.0.0`.
+2. The Temporal TypeScript SDK recommends using ESM. In order to test ESM modules, you need to run Jest with the `NODE_OPTIONS` environment variable set to `--experimental-vm-modules`, for example: `NODE_OPTIONS=--experimental-vm-modules jest`.
```

Comment:
We don't recommend this, it's up to the users to decide.
ESM is still pretty new and is still not the recommended method in Node.js.